bindings/capacitorjs/android/src/main/java/org/iota/IotaWalletMobile/IotaWalletMobile.java

```diff
@@ -0,0 +1,190 @@
+            // This is irrelevant for the Java side, but required on the Swift and JS side
+            Integer messageHandlerPointer = 1;
+            ret.put("messageHandler", messageHandlerPointer);
+            call.resolve(ret);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @PluginMethod()
+    public void sendMessage(final PluginCall call) {
+        try {
+            if (!call.getData().has("message")) {
+                call.reject("message is required");
+            }
+            String message = call.getString("message");
+            if (message == null) {
+                return;
+            }
+
+            JsonElement element = JsonParser.parseString(message);
+            JsonObject jsonObject = element.getAsJsonObject();
+            WalletCommand walletCommand;
+            if (jsonObject.has("payload")) {
+                walletCommand = new WalletCommand(
+                        jsonObject.get("cmd").getAsString(),
+                        jsonObject.get("payload")
+                );
+            }
+            else {
+                walletCommand = new WalletCommand(jsonObject.get("cmd").getAsString());
+
+            }
+            JsonElement jsonResponse = callBaseApi(walletCommand);
+            JSObject ret = new JSObject();
+            if (jsonResponse != null) {
+                Log.d("jsonResponse", jsonResponse.toString());
+                JsonObject clientResponse = new JsonObject();
+                clientResponse.addProperty("type", jsonObject.get("cmd").getAsString());
+                clientResponse.add("payload", jsonResponse);
+                ret.put("result", clientResponse.toString());
+            } else {
+                ret.put("result", "Ok");
+            }
+            call.resolve(ret);
+        } catch (Exception ex) {
+            call.reject(ex.getMessage() + Arrays.toString(ex.getStackTrace()));
+            Log.d("sendMessage Error", ex.getMessage() + Arrays.toString(ex.getStackTrace()));
+        }
+    }
+
+    @PluginMethod(returnType = PluginMethod.RETURN_CALLBACK)
+    public void listen(final PluginCall call) throws WalletException, JSONException {
+        // if (!call.getData().has("eventTypes")) {
+        //     call.reject("eventTypes and event are required");
+        // }
+        // JSArray eventTypes = call.getArray("eventTypes");
+        // try {
+        //     wallet.listen(new EventListener() {
+        //         @Override
+        //         public void receive(Event event) {
+        //             JSObject walletResponse = new JSObject();
+        //             walletResponse.put("result", event.getEvent().toString());
+        //             call.resolve(walletResponse);
+        //         }
+        //     }, WalletEventType.valueOf(eventTypes.getString(0)));
+        // } catch (WalletException e) {
+        //     e.printStackTrace();
+        // }
+        // call.setKeepAlive(true);
```

Comment:
Is this all working as expected without the commented code? If so it should be removed

Reply:
thanksm this is unhidden in the other PR, now we have working the listen method